suzy/scripts/articleView.js

```diff
@@ -0,0 +1,132 @@
+      if ($(`#category-filter option[value="${val}"]`).length === 0) {
+        $('#category-filter').append(optionTag);
+      }
+    }
+  });
+};
+
+articleView.handleAuthorFilter = () => {
+  $('#author-filter').on('change', function() {
+    if ($(this).val()) {
+      $('article').hide();
+      $(`article[data-author="${$(this).val()}"]`).fadeIn();
+    } else {
+      $('article').fadeIn();
+      $('article.template').hide();
+    }
+    $('#category-filter').val('');
+  });
+};
+
+articleView.handleCategoryFilter = () => {
+  $('#category-filter').on('change', function() {
+    if ($(this).val()) {
+      $('article').hide();
+      $(`article[data-category="${$(this).val()}"]`).fadeIn();
+    } else {
+      $('article').fadeIn();
+      $('article.template').hide();
+    }
+    $('#author-filter').val('');
+  });
+};
+
+articleView.handleMainNav = () => {
+  $('nav').on('click', '.tab', function(e) {
+    e.preventDefault();
+    $('.tab-content').hide();
+    $(`#${$(this).data('content')}`).fadeIn();
+  });
+
+  $('nav .tab:first').click();
+};
+
+articleView.setTeasers = () => {
+  $('.article-body *:nth-of-type(n+2)').hide();
+  $('article').on('click', 'a.read-on', function(e) {
+    e.preventDefault();
+    if ($(this).text() === 'Read on →') {
+      $(this).parent().find('*').fadeIn();
+      $(this).html('Show Less &larr;');
+    } else {
+      $('body').animate({
+        scrollTop: ($(this).parent().offset().top)
+      },200);
+      $(this).html('Read on &rarr;');
+      $(this).parent().find('.article-body *:nth-of-type(n+2)').hide();
+    }
+  });
+};
+
+// COMMENT: Where is this function called? Why?
+// new.html
+articleView.initNewArticlePage = () => {
+  // TODO: Ensure the main .tab-content area is revealed. We might add more tabs later or otherwise edit the tab navigation.
+
+
+  // TODO: The new articles we create will be copy/pasted into our source data file.
+  // Set up this "export" functionality. We can hide it for now, and show it once we have data to export.
+
+  $('#article-json').on('focus', function(){
+    this.select();
+  });
+
+  // TODO: Add an event handler to update the preview and the export field if any inputs change.
+  $('form').on('input',() => {
+    articleView.create();
+  })
+
+};
+
+articleView.create = () => {
+  // TODO: Set up a variable to hold the new article we are creating.
+  // Clear out the #articles element, so we can put in the updated preview
+  let newArticle = newArticle;
+  $('#articles').empty();
+
+  // TODO: Instantiate an article based on what's in the form fields:
+  new Article({
+    title: $('#title').val(),
+    author: $('#author').val(),
+    authorUrl: $('#authorUrl').val(),
+    category: $('category').val(),
+    body: $('body').val(),
+    publishedOn: new Date()
+
+  })
+  // TODO: Use our interface to the Handblebars template to put this new article into the DOM:
+  $('articles').append(newArticle.toHtml);
+
```

Comment:
*Hint* check out JSON.stringify() to convert your new article to html and use your handlebars template to render a preview on the page.